src/sentry/uptime/detectors/hostname_extraction.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from urllib.parse import urlsplit
+
+from django.core.exceptions import ValidationError
+from django.core.validators import URLValidator, validate_ipv46_address
+from tldextract import TLDExtract
+
+if TYPE_CHECKING:
+    pass
+
+url_validator = URLValidator()
+
+# Configure TLDExtract to not make external calls and just use a local snapshot of tld data
+extractor = TLDExtract(
+    cache_dir=None,
+    suffix_list_urls=(),
+    fallback_to_snapshot=True,
+)
```

Comment:
Should we expand on this comment at all with _why_ we don't want it to make external requests?